Add offset focus ring for buttons, vertical tabs etc

niik opened the pull request:
<!--
What GitHub Desktop issue does this PR address? (for example, #1234)
-->

xref: https://github.com/github/accessibility-audits/issues/3288

## Description
<!--
A summary of the changes made along with any other information that would be helpful to a reviewer such as potential tradeoffs or alternative approaches you considered.
-->
This addresses an accessibility issue where the focus ring (or lack of really) for our primary buttons doesn't meet the required contrast. Prior to this PR we only tweaked the border color slightly when a button received keyboard focus. In order to address this I'm changing buttons as well as vertical tabs to use offset focus rings and I expect that we'll find more places to leverage this pattern going forward.

### Screenshots

<!--
If this PR touches the UI layer of the app, please include screenshots or animated gifs to show the changes.
-->

| Before | After |
|--------|--------|
| <img width="240" alt="image" src="https://user-images.githubusercontent.com/634063/224118137-a8cdd77d-9bae-4336-aaa6-e96b495fb414.png"> | <img width="243" alt="image" src="https://user-images.githubusercontent.com/634063/224118277-b3b06b52-c070-46ec-8922-34ddfd9baef9.png"> |
| <img width="179" alt="image" src="https://user-images.githubusercontent.com/634063/224118350-24ce2b0d-7abd-435e-96eb-780112ae1b59.png"> | <img width="172" alt="image" src="https://user-images.githubusercontent.com/634063/224118402-90c5beee-cb58-40ce-8fcd-39631c96a038.png"> |
| <img width="67" alt="image" src="https://user-images.githubusercontent.com/634063/224118446-1316d723-fe68-48e3-838b-8730b1cf6b2b.png"> | <img width="74" alt="image" src="https://user-images.githubusercontent.com/634063/224118498-e790eb65-7c42-445c-a143-1dae684a0553.png"> | 

## Release notes

<!--
You can leave this blank if you're not sure.
If you don't believe this PR needs to be mentioned in the release notes, write "Notes: no-notes".
-->

Notes:

niik committed:
Increase border contrast on interactive buttons and text boxes

niik committed:
Skip border effect on secondary button hover

niik committed:
Make button hover state border color themable

niik committed:
Offset focus outline for buttons, vertical tabs

sergiou87 reviewed (commented):
Looks good overall! I only noticed something annoying in the commit button when its text is too long and is truncated:

![Kapture 2023-03-10 at 11 21 39](https://user-images.githubusercontent.com/1083228/224291299-c83a8164-ef57-44a7-99fa-26961671b2c8.gif)

As you can see the text moves slightly. It's a minor thing that didn't happen before this PR. I'm happy to approve it as it is, let me know what you want to do about it! 😄

niik committed:
Don't remove border on primary button focus

We have a border here so that the primary buttons and non-primary take have the same amount of space within them

niik committed:
A little extra spacing looks better

niik committed:
We don't need the alt focus now that the outline is offset

niik committed:
A little more spacing around the dialog close button focus outline

niik commented:
> As you can see the text moves slightly. It's a minor thing that didn't happen before this PR. I'm happy to approve it as it is, let me know what you want to do about it! 😄

Great spot! 😍 I thought removing the border was safe but it wasn't and we can't remove it entirely because we want the primary and secondary button styles to have the same internal dimensions so that text line up when placed next to each other. I sorted that out and fixed the focus color on vertical tabs and added a little extra spacing to make things look better.

sergiou87 reviewed (approved):
Thanks for those changes! It works perfectly now 🙌
